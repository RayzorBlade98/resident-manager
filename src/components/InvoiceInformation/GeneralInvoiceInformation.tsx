import React from 'react';
import { Invoice } from '_/types/invoice';

interface GeneralInvoiceInformationProps {
  /**
   * Invoice for which the information should be displayed
   */
  invoice: Invoice;
}

/**
 * Component that displays general information about an invoice
 */
function GeneralInvoiceInformation(
  props: GeneralInvoiceInformationProps,
): JSX.Element {
  return (
    // eslint-disable-next-line max-len
    <p>{`${props.invoice.start.month} ${props.invoice.start.year} - ${props.invoice.end.month} ${props.invoice.end.year}`}</p>
  );
}

export default GeneralInvoiceInformation;
