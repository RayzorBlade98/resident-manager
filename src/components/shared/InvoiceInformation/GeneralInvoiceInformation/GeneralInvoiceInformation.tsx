import { Button } from '@mui/material';
import React from 'react';
import Invoice from '_/models/invoice/invoice';

interface GeneralInvoiceInformationProps {
  /**
   * Invoice for which the general information should be displayed
   */
  invoice: Invoice;
}

/**
 * Component that displays general information about an invoice
 */
function GeneralInvoiceInformation(
  props: GeneralInvoiceInformationProps,
): JSX.Element {
  const invoice = props.invoice;
  return (
    <>
      <p>{`${invoice.start.toString()} - ${invoice.end.toString()}`}</p>
      <Button
        onClick={() => {
          void window.ipcAPI.generateInvoicePdfs(invoice);
        }}
      >
        Pdf generieren
      </Button>
    </>
  );
}

export default GeneralInvoiceInformation;
