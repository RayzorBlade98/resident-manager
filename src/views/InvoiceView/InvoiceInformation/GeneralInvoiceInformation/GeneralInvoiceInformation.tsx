import React from 'react';
import { useRecoilValue } from 'recoil';
import { invoiceViewSelectedInvoiceSelector } from '../../states/invoice_view_state';
import { Invoice } from '_/types/invoice';

/**
 * Component that displays general information about an invoice
 */
function GeneralInvoiceInformation(): JSX.Element {
  const selectedInvoice = useRecoilValue(
    invoiceViewSelectedInvoiceSelector,
  ) as Invoice;

  return (
    // eslint-disable-next-line max-len
    <p>{`${selectedInvoice.start.month} ${selectedInvoice.start.year} - ${selectedInvoice.end.month} ${selectedInvoice.end.year}`}</p>
  );
}

export default GeneralInvoiceInformation;
