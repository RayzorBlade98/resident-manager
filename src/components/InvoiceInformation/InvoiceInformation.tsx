import { StyleAttribute } from 'glamor';
import {
  MDBTabs,
  MDBTabsContent,
  MDBTabsItem,
  MDBTabsLink,
  MDBTabsPane,
} from 'mdb-react-ui-kit';
import React, { useState } from 'react';
import GeneralInvoiceInformation from './GeneralInvoiceInformation';
import { Invoice } from '_/types/invoice';

/**
 * Enum containing all tabs of this component
 */
enum InvoiceTab {
  General = 'Information',
}

interface InvoiceInformationProps {
  /**
   * Invoice for which the information should be displayed
   */
  invoice: Invoice;

  /**
   * Style that should be applied to the container of the component. (Optional)
   */
  containerStyle: StyleAttribute;
}

/**
 * Component that displays information of the invoice in different tabs.
 */
function InvoiceInformation(props: InvoiceInformationProps): JSX.Element {
  // Currently selected tab
  const [activeTab, setActiveTab] = useState<InvoiceTab>(InvoiceTab.General);

  /**
   * Handles the tab selection
   * @param tab selected tab
   */
  function onClickTab(tab: InvoiceTab): void {
    if (tab === activeTab) return;
    setActiveTab(tab);
  }

  /**
   * Returns the content of the currently active tab
   */
  function getTabContent(): JSX.Element {
    switch (activeTab) {
      case InvoiceTab.General:
        return <GeneralInvoiceInformation invoice={props.invoice} />;
      default:
        throw new Error(`Tab ${activeTab} is not implemented`);
    }
  }

  return (
    <div {...props.containerStyle}>
      <MDBTabs fill className="mb-3">
        {Object.values(InvoiceTab).map((tab: InvoiceTab) => (
          <MDBTabsItem>
            <MDBTabsLink
              onClick={() => onClickTab(tab)}
              active={activeTab === tab}
            >
              {tab}
            </MDBTabsLink>
          </MDBTabsItem>
        ))}
      </MDBTabs>

      <MDBTabsContent>
        <MDBTabsPane show>
          <div>{getTabContent()}</div>
        </MDBTabsPane>
      </MDBTabsContent>
    </div>
  );
}

export default InvoiceInformation;
