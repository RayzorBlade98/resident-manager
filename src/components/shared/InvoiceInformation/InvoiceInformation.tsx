import InfoIcon from '@mui/icons-material/Info';
import { Box, Tab, Tabs } from '@mui/material';
import React, { useState } from 'react';
import GeneralInvoiceInformation from './GeneralInvoiceInformation/GeneralInvoiceInformation';
import Invoice from '_/models/invoice/invoice';

/**
 * Enum containing all tabs of this component
 */
enum InvoiceTab {
  General = 0,
}

const styles = {
  tabsBox: { borderBottom: 1, borderColor: 'divider' },
};

interface InvoiceInformationProps {
  /**
   * Invoice for which the information should be displayed
   */
  invoice: Invoice;
}

/**
 * Component that displays information of the invoice in different tabs.
 */
function InvoiceInformation(props: InvoiceInformationProps): JSX.Element {
  const [activeTab, setActiveTab] = useState<InvoiceTab>(InvoiceTab.General);

  return (
    <>
      <Box sx={styles.tabsBox}>
        <Tabs
          variant="fullWidth"
          value={activeTab}
          onChange={(_, tab: InvoiceTab) => setActiveTab(tab)}
        >
          <Tab icon={<InfoIcon />} iconPosition="start" label="Informationen" />
        </Tabs>
      </Box>
      <div hidden={activeTab !== InvoiceTab.General}>
        <GeneralInvoiceInformation invoice={props.invoice} />
      </div>
    </>
  );
}

export default InvoiceInformation;
