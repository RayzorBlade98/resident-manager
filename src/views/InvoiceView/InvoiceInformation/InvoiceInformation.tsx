import InfoIcon from '@mui/icons-material/Info';
import { Box, Tab, Tabs } from '@mui/material';
import React, { useState } from 'react';
import GeneralInvoiceInformation from './GeneralInvoiceInformation/GeneralInvoiceInformation';

/**
 * Enum containing all tabs of this component
 */
enum InvoiceTab {
  General = 0,
}

const styles = {
  tabsBox: { borderBottom: 1, borderColor: 'divider' },
};

/**
 * Component that displays information of the invoice in different tabs.
 */
function InvoiceInformation(): JSX.Element {
  const [activeTab, setActiveTab] = useState<InvoiceTab>(InvoiceTab.General);

  return (
    <>
      <Box sx={styles.tabsBox}>
        <Tabs
          variant="fullWidth"
          value={activeTab}
          onChange={(_, tab: InvoiceTab) => setActiveTab(tab)}
        >
          <Tab icon={<InfoIcon />} iconPosition="start" label="Informationen" />
        </Tabs>
      </Box>
      <div hidden={activeTab !== InvoiceTab.General}>
        <GeneralInvoiceInformation />
      </div>
    </>
  );
}

export default InvoiceInformation;
