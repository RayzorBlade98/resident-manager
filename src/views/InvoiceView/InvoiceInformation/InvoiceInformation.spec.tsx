import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import * as GeneralInvoiceInformation from './GeneralInvoiceInformation/GeneralInvoiceInformation';
import InvoiceInformation from './InvoiceInformation';

describe('InvoiceInformation', () => {
  beforeAll(() => {
    jest
      .spyOn(GeneralInvoiceInformation, 'default')
      .mockReturnValue(<div>GeneralInvoiceInformation</div>);
  });

  test.each([['GeneralInvoiceInformation', 0]])(
    'should match snapshot (selected %s tab)',
    (tabName: string, tabId: number) => {
      // Arrange
      const renderResult = render(<InvoiceInformation />);

      // Act
      const tab = renderResult.getAllByRole('tab').at(tabId)!;
      fireEvent.click(tab);

      // Assert
      expect(renderResult.container).toMatchSnapshot();
    },
  );
});
