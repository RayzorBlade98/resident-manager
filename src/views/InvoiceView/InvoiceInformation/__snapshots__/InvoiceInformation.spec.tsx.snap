// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceInformation should match snapshot (selected GeneralInvoiceInformation tab) 1`] = `
<div>
  <div
    class="MuiBox-root css-1gsv261"
  >
    <div
      class="MuiTabs-root css-1ujnqem-MuiTabs-root"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
        style="overflow: hidden; margin-bottom: 0px;"
      >
        <div
          class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-labelIcon MuiTab-textColorPrimary MuiTab-fullWidth Mui-selected css-1nppm96-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTab-iconWrapper css-i4bv87-MuiSvgIcon-root"
              data-testid="InfoIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
              />
            </svg>
            Informationen
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
  </div>
  <div>
    <div>
      GeneralInvoiceInformation
    </div>
  </div>
</div>
`;
